src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java

```diff
@@ -297,6 +311,17 @@ public Deployment.DeploymentStage determineDeploymentStage(BootstrapManager boot
             }
             return KERNEL_ACTIVATION;
         } else if (getBrokenDir().toFile().exists()) {
+            try {
+                Path rollbackBootstrapTasks = deploymentDirectoryManager.getRollbackBootstrapTaskFilePath();
+                if (rollbackBootstrapTasks.toFile().exists()) {
+                    bootstrapManager.loadBootstrapTaskList(rollbackBootstrapTasks);
+                    if (bootstrapManager.hasNext()) {
+                        return ROLLBACK_BOOTSTRAP;
+                    }
+                }
+            } catch (IOException e) {
+                logger.atError().setCause(e).log("Bootstrap-on-rollback task list not found or not readable");
```

Comment:
If it wasn't found you should not have any exception, so just saying unreadable is fine.

Reply:
That's true, I'll just log that it was unreadable.